pkg/kratos/service.go

```diff
@@ -163,6 +167,64 @@ func (s *Service) GetFlowError(ctx context.Context, id string) (*kClient.FlowErr
 	return flowError, resp.Cookies(), nil
 }
 
+func (s *Service) CheckAllowedProvider(ctx context.Context, loginFlow *kClient.LoginFlow, updateFlowBody *kClient.UpdateLoginFlowBody) (bool, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.Service.CheckAllowedProvider")
+	defer span.End()
+
+	provider := updateFlowBody.UpdateLoginFlowWithOidcMethod.Provider
+	clientName := loginFlow.Oauth2LoginRequest.Client.GetClientName()
+
+	allowed, err := s.authz.Check(ctx, fmt.Sprintf("app:%s", clientName), "allowed_access", fmt.Sprintf("provider:%s", provider))
+	if err != nil {
+		// We return false if an error occurs, this means that a fail on the authorization server will block our service
+		return false, err
+	}
```

Comment:
i would expect that allowed is set to false if err is not nil and in that case a simple 

```
s.authz.Check(ctx, fmt.Sprintf("app:%s", clientName), "allowed_access", fmt.Sprintf("provider:%s", provider))
```

main reason is a bool defaults to false so it should be quite straightforward